libraries/networking/src/udt/Connection.cpp

```diff
@@ -825,30 +825,35 @@ void PendingReceivedMessage::enqueuePacket(std::unique_ptr<Packet> packet) {
         return;
     }
 
+    auto sequenceNumber = packet->getSequenceNumber();
+
     if (packet->getPacketPosition() == Packet::PacketPosition::FIRST) {
         _hasFirstSequenceNumber = true;
-        _firstSequenceNumber = packet->getSequenceNumber(); 
+        _firstSequenceNumber = sequenceNumber;
     } else if (packet->getPacketPosition() == Packet::PacketPosition::LAST) {
         _hasLastSequenceNumber = true;
-        _lastSequenceNumber = packet->getSequenceNumber(); 
+        _lastSequenceNumber = sequenceNumber;
     } else if (packet->getPacketPosition() == Packet::PacketPosition::ONLY) {
         _hasFirstSequenceNumber = true;
         _hasLastSequenceNumber = true;
-        _firstSequenceNumber = packet->getSequenceNumber(); 
-        _lastSequenceNumber = packet->getSequenceNumber(); 
+        _firstSequenceNumber = sequenceNumber;
+        _lastSequenceNumber = sequenceNumber;
     }
 
-    _packets.push_back(std::move(packet));
+    // Insert into the packets list in sorted order. Because we generally expect to receive packets in order, begin
+    // searching from the end of the list.
+    auto it = _packets.rbegin();
+    for (auto rend = _packets.rend(); it != rend; ++it) {
+        if (sequenceNumber > (*it)->getSequenceNumber()) {
+            break;
+        }
+    }
+    _packets.insert(it.base(), std::move(packet));
```

Comment:
Nice, I didn't know the `base` method. Can be useful.
You could probably use `std::find` above.